keanu-python/tests/test_vertex_operations.py

```diff
@@ -135,126 +138,168 @@ def test_can_do_less_than_or_equal_to_with_vertex_on_rhs(lhs: Union[Vertex, nump
-                               expected_result: numpy_types) -> None:
-    result: Union[Vertex, np._ArrayLike[Any]] = lhs * rhs
-    assert isinstance(result, Vertex)
-    assert (result.get_value() == expected_result).all()
-    result = rhs * lhs
-    assert isinstance(result, Vertex)
-    assert (result.get_value() == expected_result).all()
+def test_can_do_addition(lhs_constructor: Callable, rhs_constructor: Callable, lhs: arithmetic_op_value_types,
+                         rhs: arithmetic_op_value_types, expected_result: arithmetic_op_value_types) -> None:
+    lhs_casted = lhs_constructor(lhs)
+    rhs_casted = rhs_constructor(rhs)
+
+    if result_should_be_vertex(lhs_casted, rhs_casted):
+        result: arithmetic_op_value_types = lhs_casted + rhs_casted
+        assert_is_correct_vertex_type_and_expected_value(lhs_casted, rhs_casted, result, expected_result)
+
+        result = rhs_casted + lhs_casted
+        assert_is_correct_vertex_type_and_expected_value(lhs_casted, rhs_casted, result, expected_result)
 
 
 # yapf: disable
+@pytest.mark.parametrize("lhs_constructor", types_to_test)
+@pytest.mark.parametrize("rhs_constructor", types_to_test)
 @pytest.mark.parametrize("lhs, rhs, expected_result", [
-    (Const(np.array([15., 10.])), Const(np.array([2., 4.])), np.array([7.5, 2.5 ])),
-    (Const(np.array([15., 10.])),       np.array([2., 4.]) , np.array([7.5, 2.5 ])),
-    (Const(np.array([15., 10.])),                 2.       , np.array([7.5, 5.  ])),
+    (       np.array([10, 20]),       np.array([1, 2]),   np.array([9, 18])),
+    (       np.array([10, 20]),                      2,   np.array([8, 18])),
+    (                        2,     np.array([10, 20]), np.array([-8, -18])),
+    (                        3,                      2,                   1),
 ])
 # yapf: enable
-def test_can_do_division(lhs: Vertex, rhs: Union[Vertex, numpy_types, float], expected_result: numpy_types) -> None:
-    result: Union[Vertex, np._ArrayLike[Any]] = lhs / rhs
-    assert isinstance(result, Vertex)
-    assert (result.get_value() == expected_result).all()
-    result = rhs / lhs
-    assert isinstance(result, Vertex)
-    assert (result.get_value() == 1. / expected_result).all()
+def test_can_do_subtraction(lhs_constructor: Callable, rhs_constructor: Callable, lhs: arithmetic_op_value_types,
+                            rhs: arithmetic_op_value_types, expected_result: arithmetic_op_value_types) -> None:
+    lhs_casted = lhs_constructor(lhs)
+    rhs_casted = rhs_constructor(rhs)
+
+    if result_should_be_vertex(lhs_casted, rhs_casted):
+        result: arithmetic_op_value_types = lhs_casted - rhs_casted
+        assert_is_correct_vertex_type_and_expected_value(lhs_casted, rhs_casted, result, expected_result)
+
+        result = rhs_casted - lhs_casted
+        assert_is_correct_vertex_type_and_expected_value(lhs_casted, rhs_casted, result, -expected_result)
```

Comment:
why the minus sign on `-expected_result`?

Reply:
This is the rhs - lhs which is testing the other way round so the expected answer is negated.